clinica/pipelines/statistics_volume/statistics_volume_utils.py

```diff
@@ -238,72 +379,104 @@ def write_covariate_lines(m_file_to_write_in, covar_number, covar_name, covar_va
+    If launch with spm standalone, the line 'spm_jobman('run', matlabbatch)' must be removed because unnecessary
 
-    Args:
-        m_file: (str) path to Matlab m file
+    Parameters
+    ----------
+    m_file: str
+        Path to Matlab m file
 
-    Returns:
-        output_mat_file: (str) path to the SPM.mat file needed in SPM analysis
+    Returns
+    -------
+    output_mat_file: str
+        Path to the SPM.mat file needed in SPM analysis
     """
-    import platform
-    from os import system
-    from os.path import abspath, basename, dirname, isfile, join
 
-    from nipype.interfaces.matlab import MatlabCommand, get_matlab_command
+    from os.path import abspath, dirname, isfile, join
+    from pathlib import Path
 
-    import clinica.pipelines.statistics_volume.statistics_volume_utils as utls
     from clinica.utils.spm import spm_standalone_is_available
 
-    assert isinstance(m_file, str), "[Error] Argument must be a string"
-    if not isfile(m_file):
+    if not Path(m_file).exists():
         raise FileNotFoundError("[Error] File " + m_file + "does not exist")
-    assert m_file[-2:] == ".m", (
-        "[Error] " + m_file + " is not a Matlab file (extension must be .m)"
-    )
+    if Path(m_file).suffix != ".m":
+        raise ValueError(
+            f"[Error] {m_file} is not a Matlab file (extension must be .m)"
+        )
 
     # Generate command line to run
     if spm_standalone_is_available():
-        utls.delete_last_line(m_file)
-        # SPM standalone must be run directly from its root folder
-        if platform.system().lower().startswith("darwin"):
-            # Mac OS
-            cmdline = (
-                "cd $SPMSTANDALONE_HOME && ./run_spm12.sh $MCR_HOME batch " + m_file
-            )
-        elif platform.system().lower().startswith("linux"):
-            # Linux OS
-            cmdline = "$SPMSTANDALONE_HOME/run_spm12.sh $MCR_HOME batch " + m_file
-        else:
-            raise SystemError("Clinica only support Mac OS and Linux")
-        system(cmdline)
+        run_matlab_script_with_spm_standalone(m_file)
     else:
-        MatlabCommand.set_default_matlab_cmd(get_matlab_command())
-        matlab = MatlabCommand()
-        if platform.system().lower().startswith("linux"):
-            matlab.inputs.args = "-nosoftwareopengl"
-        matlab.inputs.paths = dirname(m_file)
-        matlab.inputs.script = basename(m_file)[:-2]
-        matlab.inputs.single_comp_thread = False
-        matlab.inputs.logfile = abspath("./matlab_output.log")
-        matlab.run()
+        run_matlab_script_with_matlab(m_file)
     output_mat_file = abspath(join(dirname(m_file), "..", "2_sample_t_test", "SPM.mat"))
     if not isfile(output_mat_file):
         raise RuntimeError("Output matrix " + output_mat_file + " was not produced")
     return output_mat_file
 
 
-def delete_last_line(filename):
+def run_matlab_script_with_matlab(m_file: str) -> None:
```

Comment:
It's a Path object that you expect:

```suggestion
def run_matlab_script_with_matlab(m_file: Path) -> None:
```